src/app/login/login.page.ts

```diff
@@ -5,36 +5,38 @@ import { LoadingController } from '@ionic/angular';
 @Component({
   selector: 'app-login',
   templateUrl: './login.page.html',
-  styleUrls: ['./login.page.scss'],
+  styleUrls: ['./login.page.scss']
 })
 export class LoginPage implements OnInit {
+  errorMessage: string;
 
-  constructor(private authService: AuthenticationService, public loadingController: LoadingController) { }
+  constructor(private authService: AuthenticationService, private loadingController: LoadingController) {}
 
   async ngOnInit() {
+    console.log('ngOnInit');
     // If coming back after logging into Auth0,
     // and using CURRENT Implicit (web) Login
     if (window.location.hash) {
       const loadingIndicator = await this.showLoadingIndictator();
-
-      // Pass it to Auth Connect
-      await this.authService.callback(window.location.href, loadingIndicator);
+      try {
+        await this.authService.handleCallback(window.location.href);
+      } catch (e) {
+        this.errorMessage = e.message;
+      } finally {
+        loadingIndicator.dismiss();
+      }
     }
-   }
+  }
 
   async login() {
```

Comment:
The loading indicator was removed here because if you Push "Done" or "Cancel" from the Auth0 login without logging in, there isn't an event currently to latch into to dismiss the indicator. Ideally, the login() would resolve or reject in this case, but it does not.